Wikipedia/Code/TalkPageTopicComposeViewController+TalkPageFormattingToolbar.swift

```diff
@@ -26,20 +28,33 @@ extension TalkPageTopicComposeViewController: TalkPageFormattingToolbarViewDeleg
                 if let newText = bodyTextView.text(in: newSelectedRange) {
                     if newText.contains("[") || newText.contains("]") {
                         doesLinkExist = true
+
                     } else {
                         doesLinkExist = false
                     }
                 }
             }
 
-            guard let link = Link(page: text, label: text, exists: doesLinkExist) else {
+            let index = text.firstIndex(of: "|") ?? text.endIndex
+            let beggining = text[..<index]
+
+            let ending = text[index...]
+
+            let newSearchTerm = String(beggining)
+            let newLabel = String(ending.dropFirst())
+
+            let linkText = doesLinkExist ? newSearchTerm : text
+            let labelText = doesLinkExist ? newLabel : text
```

Comment:
I notice this chunk of logic is missing in the other `didSelectInsertLink` method - so there's a bug in the reply textview that is fixed on this new topic textview here. Can you share some of this logic in your new UITextView+FormattingToolbarExtension.swift class?